projects/npm-tools/packages/npm-scripts/src/jsp/getTaglibInvocations.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * SPDX-FileCopyrightText: © 2019 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const Lexer = require('./Lexer');
+
+const DEFAULT_OPTIONS = {
+	ELEnabled: true,
+};
+
+function lex(source, options = {}) {
+	const {ELEnabled} = {
+		...DEFAULT_OPTIONS,
+		...options,
+	};
+
+	const lexer = new Lexer((api) => {
+		const {consume, fail, match, meta, peek, repeat, sequence, token} = api;
+
+		meta.set('ELEnabled', !!ELEnabled);
+		meta.commit();
+
+		const {
+			ATTRIBUTE,
+			CHAR,
+			CUSTOM_ACTION,
+			CUSTOM_ACTION_START,
+			SPACE,
+		} = require('./jspDSL')(api);
+
+		return () => {
+			if (peek(sequence(CUSTOM_ACTION_START, CUSTOM_ACTION))) {
+				consume(CUSTOM_ACTION_START);
+
+				return token('TAG', consume(CUSTOM_ACTION));
+			}
+			else if (peek(SPACE)) {
+				consume(SPACE);
+
+				return token('SPACE', '·');
+			}
+			else if (peek(ATTRIBUTE)) {
+				return token('ATTRIBUTE', consume());
+			}
+			else if (peek(match('>'))) {
+				consume(repeat(CHAR));
+
+				return token('REST', '·');
+			}
+			else {
+				fail('Failed to consume all input');
+			}
+		};
+	});
+
+	return [...lexer.lex(source)];
+}
+
+module.exports = function (jsp, ...tags) {
+	const invocations = [];
+
+	for (const tag of tags) {
+		let from = 0;
+
+		while ((from = jsp.indexOf(`<${tag}`, from)) != -1) {
```

Comment:
Do `!==` here instead of `!=` because the latter [will be a lint error](https://github.com/liferay/liferay-frontend-projects/issues/14) once we turn on the rule.